draft-ietf-tls-tls13.md

```diff
@@ -3107,101 +3267,45 @@ Structure of this message:
 
 The verify_data value is computed as follows:
 
-verify_data
-: HMAC(finished_secret, finished_label + '\0' + handshake_hash)
-  where HMAC {{RFC2104}} uses the Hash algorithm for the handshake.
-  See {{the-handshake-hash}} for the definition of handshake_hash.
+~~~
+verify_data =
+  HMAC(finished_key, finished_label + '\0' + Handshake Context +
+                     Certificate + CertificateVerify)
+~~~
+
+Where:
+
+HMAC
+: HMAC {{RFC2104}} using the Hash algorithm for the handshake.
+As noted above: the HMAC input can generally be implemented by a running
+hash.
 
 finished_label
 : For Finished messages sent by the client, the string
   "client finished".  For Finished messages sent by the server,
-  the string "server finished".
-{:br }
+  the string "server finished" [[OPEN ISSUE: Do we still need these?
+  labels? Should we expand them to indicate which phase we are
+  sending the messages in?]]
```

Comment:
If you do keep the labels, then something needs to change.  Having both "server finished" and "server_finished" is going to be quite confusing.